HomeWork2/Clean Architecture/Catalog Service/CatalogService.BLL/Interfaces/Services/ProductService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Application.Common;
+using Application.Interfaces.Repositories;
+using AutoMapper;
+using Catalog_Service;
+using Domain;
+
+namespace Application.Interfaces.Services;
+
+public class ProductService : IProductService
+{
+    private readonly IGenericRepository<Product> repository;
+    private readonly Mapper mapper;
+
+    public ProductService(IGenericRepository<Product> genericRepository)
+    {
+        repository = genericRepository;
+        mapper = MapperConfig.InitializeAutomapper();
```

Comment:
you have to inject IMapper and don't initilize it on BLL level

Reply:
changed